blockchain/utxoviewpoint.go

```diff
@@ -533,25 +533,25 @@ func (view *UtxoViewpoint) fetchUtxosMain(db database.DB, outpoints map[wire.Out
 // fetchUtxos loads the unspent transaction outputs for the provided set of
 // outputs into the view from the database as needed unless they already exist
 // in the view in which case they are ignored.
-func (view *UtxoViewpoint) fetchUtxos(db database.DB, outpoints map[wire.OutPoint]struct{}) error {
+func (view *UtxoViewpoint) fetchUtxos(db database.DB, outpoints []wire.OutPoint) error {
 	// Nothing to do if there are no requested outputs.
 	if len(outpoints) == 0 {
 		return nil
 	}
 
 	// Filter entries that are already in the view.
-	neededSet := make(map[wire.OutPoint]struct{})
-	for outpoint := range outpoints {
+	needed := make([]wire.OutPoint, 0, len(outpoints))
+	for _, outpoint := range outpoints {
 		// Already loaded into the current view.
 		if _, ok := view.entries[outpoint]; ok {
 			continue
 		}
 
-		neededSet[outpoint] = struct{}{}
+		needed = append(needed, outpoint)
```

Comment:
I think this is the classic example of where things go wrong with Golang loop variables. Should use index based access to the `outpoint`.

Reply:
Addressed with the latest push.